libraries/chain/include/eosio/chain/webassembly/eos-vm-oc.hpp

```diff
@@ -42,225 +58,14 @@ class eosvmoc_runtime : public eosio::chain::wasm_runtime_interface {
       eosvmoc::memory mem;
 };
 
-/**
- * validate an in-wasm-memory array
- * @tparam T
- */
-template<typename T>
-inline void* array_ptr_impl (size_t ptr, size_t length)
-{
-   constexpr int cb_full_linear_memory_start_segment_offset = OFFSET_OF_CONTROL_BLOCK_MEMBER(full_linear_memory_start);
-   constexpr int cb_first_invalid_memory_address_segment_offset = OFFSET_OF_CONTROL_BLOCK_MEMBER(first_invalid_memory_address);
-
-   size_t end = ptr + length*sizeof(T);
-
-   asm volatile("cmp %%gs:%c[firstInvalidMemory], %[End]\n"
-                "jle 1f\n"
-                "mov %%gs:(%[End]), %[Ptr]\n"      // invalid pointer if out of range
-                "1:\n"
-                "add %%gs:%c[linearMemoryStart], %[Ptr]\n"
-                : [Ptr] "+r" (ptr),
-                  [End] "+r" (end)
-                : [linearMemoryStart] "i" (cb_full_linear_memory_start_segment_offset),
-                  [firstInvalidMemory] "i" (cb_first_invalid_memory_address_segment_offset)
-                : "cc"
-               );
-
-
-   return (void*)ptr;
-}
-
-/**
- * validate an in-wasm-memory char array that must be null terminated
- */
-inline char* null_terminated_ptr_impl(uint64_t ptr)
-{
-   constexpr int cb_full_linear_memory_start_segment_offset = OFFSET_OF_CONTROL_BLOCK_MEMBER(full_linear_memory_start);
-   constexpr int cb_first_invalid_memory_address_segment_offset = OFFSET_OF_CONTROL_BLOCK_MEMBER(first_invalid_memory_address);
-
-   char dumpster;
-   uint64_t scratch;
-
-   asm volatile("mov %%gs:(%[Ptr]), %[Dumpster]\n"                   //probe memory location at ptr to see if valid
-                "mov %%gs:%c[firstInvalidMemory], %[Scratch]\n"      //get first invalid memory address
-                "cmpb $0, %%gs:-1(%[Scratch])\n"                     //is last byte in valid linear memory 0?
-                "je 2f\n"                                            //if so, this will be a null terminated string one way or another
-                "mov %[Ptr],%[Scratch]\n"
-                "1:\n"                                               //start loop looking for either 0, or until we SEGV
-                "inc %[Scratch]\n"
-                "cmpb $0,%%gs:-1(%[Scratch])\n"
```

Comment:
This fix needs to be preserved.

Reply:
Hmm, scary that slipped through. I was actually going to look for that and forgot. Great catch!!